### PR TITLE
Feat : Trimmed label to prevent white spaces

### LINKED ISSUE #1183: Trim label to prevent whitespaces
or other non printable chars

### PR DESCRIPTION
**Pull request by ruthishvitwit**
closes #1183 

The aim of this pr is to prevent white spaces  in labels by either silent trimming before storing  or to fail when the label is not trimmed

### CODE REVIEW

File: x/wasm/types/proposal.go [hunk 1]
```diff
@@ -208,6 +208,10 @@ func (p InstantiateContractProposal) ValidateBasic() error {
 	if p.CodeID == 0 {
 		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "code id is required")
 	}
+	//this code can be removed based on the preference between silent trimming the label or validating the label
+	if p.Label != strings.TrimSpace(p.Label) {
```
Comment (alpe) [inline x/wasm/types/proposal.go:212]: Good idea to always trim for gov proposals. So that we are safe to not reject any in flight proposals. The code can be simplified though: 
```suggestion
	p.Label != strings.TrimSpace(p.Label)
```

File: x/wasm/types/proposal.go [hunk 2]
```diff
@@ -208,6 +208,10 @@ func (p InstantiateContractProposal) ValidateBasic() error {
 	if p.CodeID == 0 {
 		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "code id is required")
 	}
+	//this code can be removed based on the preference between silent trimming the label or validating the label
```
Comment (alpe) [inline x/wasm/types/proposal.go:211]: Please do not provide options or todos in code comment. It is hard to follow up on this and it opens another chain for work.You can use the PR description or give review feedback on your own code.

A good code comment describes the motivation or adds a bigger context on why this makes sense.

File: x/wasm/types/validation.go [hunk 3]
```diff
@@ -40,6 +41,9 @@ func ValidateLabel(label string) error {
 	if len(label) > MaxLabelSize {
 		return ErrLimit.Wrapf("cannot be longer than %d characters", MaxLabelSize)
 	}
+	if label != strings.TrimSpace(label) {
+		return ErrInvalid.Wrapf("'%s'This label is not trimmed", label)
```
Comment (alpe) [inline x/wasm/types/validation.go:45]: Also very good to reject all labels with whitespaces as default behaviour. This ensures a clean state.
No need to return the original string, just the error context. For example:
```suggestion
		return ErrInvalid.Wrap("label must not start/end with whitespaces")
```

**Review by alpe — COMMENTED** [review]
Good start!
I added some comments to polish the code. Please add test cases for regression. You can find existing ones in `proposal_test.go` and `tx_test.go`

File: x/wasm/types/proposal.go (hunk 2 above)
Comment (ruthishvitwit) [reply x/wasm/types/proposal.go:211]: got it

File: x/wasm/types/proposal.go (hunk 1 above)
Comment (ruthishvitwit) [reply x/wasm/types/proposal.go:212]: do u mean
p.Label = strings.TrimSpace(p.Label)
